ddl/rollingback.go

```diff
@@ -447,12 +441,32 @@ func convertJob2RollbackJob(w *worker, d *ddlCtx, t *meta.Meta, job *model.Job)
 		if job.Error == nil {
 			job.Error = toTError(err)
 		}
-		if !job.Error.Equal(errCancelledDDLJob) {
-			job.Error = terror.GetErrClass(job.Error).Synthesize(terror.ErrCode(job.Error.Code()),
-				fmt.Sprintf("DDL job rollback, error msg: %s", terror.ToSQLError(job.Error).Message))
-		}
 		job.ErrorCount++
 
+		if errCancelledDDLJob.Equal(err) {
+			// job is normal cancelled.
+			if !job.Error.Equal(errCancelledDDLJob) {
+				job.Error = terror.GetErrClass(job.Error).Synthesize(terror.ErrCode(job.Error.Code()),
+					fmt.Sprintf("DDL job rollback, error msg: %s", terror.ToSQLError(job.Error).Message))
+			}
+		} else {
+			// job canceling meet other error.
+			//
+			// Once `convertJob2RollbackJob` meets error, the job state can't be set as `JobStateRollingback` since
+			// job state and args may not be correctly overwritten. The job will be fetched to run with the cancelling
+			// state again. So we should check the error count here.
+			if err1 := loadDDLVars(w); err1 != nil {
+				logutil.Logger(w.logCtx).Error("[ddl] load DDL global variable failed", zap.Error(err1))
+			}
+			errorCount := variable.GetDDLErrorCountLimit()
+			if job.ErrorCount > errorCount {
+				msg := fmt.Sprintf("job being converted to rollback job errors and error count beyond the limitation %d, cancelled", errorCount)
```

Comment:
```suggestion
				logutil.Logger(w.logCtx).Warn("[ddl] rollback DDL job error count exceed the limit, cancelled it now", zap.Int64("jobID", job.ID), zap.Int64("errorCountLimit", errorCount))
```

Unify the logs in `runDDL` and this function.

Reply:
addressed